docs/JLBP-0021.md

```diff
@@ -0,0 +1,28 @@
+---
+jlbp:
+  id: JLBP-21
+permalink: /JLBP-21
+---
+# Upload artifacts to Maven Central
+
+Developers expect to find artifacts in [Maven Central](https://repo1.maven.org/maven2/). All artifacts
+intended for public consumption
+should be published in the [Maven Central repository maintained by Sonatype](https://search.maven.org). Even if a project is not built with Maven,
+its artifacts can and should be published to Maven Central.
+
+From here the artifacts are mirrored to many different local and public repositories. 
+
+While Maven supports additional remote repositories,
+this requires special configuration
+in the pom.xml file. Tools other than Maven itself that 
+use the repository system do not always read this information from the 
+pom.xml files and often cannot load artifacts that are not available in Maven Central.
+
+Developers who manually search for particular artifacts usually start at
+Maven Central and sometimes don't look elsewhere.
+
+Firewalls sometimes block access to third party repositories. 
```

Comment:
I didn't find this sentence useful. Should this paragraph guide readers to take some action upon firewall problem?

Reply:
The action to take it to put things in Maven Central, not in the repositories the firewall blocks.

Reply:
I misread this. Now I know this sentence is here to support "Upload artifacts to Maven Central".

My confusion came from the fact that my previous workplace blocked Maven Central and copied only approved artifacts. So Maven Central is not immune to firewalls.

Reply:
For companies that maintains their own repositories inside their network, Maven Central is a third party repository. They sometimes block Maven Central by firewalls or HTTP proxy.

Reply:
Do you still need this line of Maven Central's immunity against firewalls? Companies may block Maven Central by firewalls or HTTP proxy. Having said that, it's easier for them to host a repository mirror inside their network when every library is hosted in Maven Central.

Reply:
> Firewalls sometimes block access to third party repositories. 

I still think this paragraph is unnecessary, because sometimes firewalls block access to Maven Central.

Reply:
I suspect it happens, but rarely. When it does, there's nothing a library developer can do about it. If the end user developer's firewall is blocking maven central, they have to fix that before they can build anything.

Reply:
Right. The existence of the restricted networks also supports having everything in Maven Central. How about saying this?

"If a library is hosted in certain repositories other than Maven Central, then it would a chore for network administrators maintaining a restricted network to setup additional mirrors or firewall rules of the repositories"

Reply:
done